core/tracker/tracker.go

```diff
@@ -97,9 +102,9 @@ func (t *Tracker) Run(ctx context.Context) error {
 	}
 }
 
-// analyseFailedDuty analyzes the events for a deadlined duty. It returns true if the duty didn't complete the sigagg component.
+// analyseDutyFailed analyzes the events for a deadlined duty. It returns true if the duty didn't complete the sigagg component.
```

Comment:
suggest not renaming since this analysis is specifically for "failed duties", the other analysis will be for "duty participation"

Reply:
clarification: `analyseFailedDuty` seems like "analysing a failed duty" which means it already assumes that the input duty is a failed one, which maynot be the case.

Shouldn't the function signature be 
`func analyseFailedDuty(core.Duty, []event) (component, string)`

instead of 

`func analyseFailedDuty(core.Duty, []event) (bool, component, string)`

Reply:
So if we have two different analysis:
 - Detect if a duty failed or succeeded and if it failed indicate the reason
 - Detect which peers participated in a duty
 
 What would you call those two different analysis and what would the signatures be?

Reply:
Probably i'd go for:

```go
// analyseDutyFailed detects if a duty failed. It returns true if the duty failed and returns the component
// where it failed and a message why.
func analyseDutyFailed(core.Duty, []event) (bool, component, string) {}

// analysePeerParticipation returns the pubkeys of the peers who participated in the duty.
func analysePeerParticipation(core.Duty, []event) []core.Pubkey {}
```

Reply:
sounds good

nit:
```
// analyseParticipation returns the share indexes of peers that participated in this duty.
func analyseParticipation(core.Duty, []event) []int {}
```